lib/rubocop/cop/mixin/hash_slice_except.rb

```diff
@@ -0,0 +1,160 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    # Common functionality for Style/HashExcept and Style/HashSlice
+    module HashSliceExcept
+      include RangeHelp
+      extend NodePattern::Macros
+
+      RESTRICT_ON_SEND = %i[reject select filter].freeze
+
+      # @!method bad_method_with_poro?(node)
+      def_node_matcher :bad_method_with_poro?, <<~PATTERN
+        (block
+          (call _ _)
+          (args
+            $(arg _)
+            (arg _))
+          {
+            $(send
+              _ {:== :!= :eql? :include?} _)
+            (send
+              $(send
+                _ {:== :!= :eql? :include?} _) :!)
+            })
+      PATTERN
+
+      # @!method bad_method_with_active_support?(node)
+      def_node_matcher :bad_method_with_active_support?, <<~PATTERN
+        (block
+          (send _ _)
+          (args
+            $(arg _)
+            (arg _))
+          {
+            $(send
+              _ {:== :!= :eql? :in? :include? :exclude?} _)
+            (send
+              $(send
+                _ {:== :!= :eql? :in? :include? :exclude?} _) :!)
+            })
+      PATTERN
+
+      def on_send(node)
+        raise NotImplementedError
+      end
```

Comment:
Can you remove this method? Mixins aren't abstract classes, and this exception doesn't have much value (and also if this mixin is later included in another cop class that doesn't define `on_send` it'll raise for every `send` node).

Reply:
Sure, but I added this because otherwise I get an offense for `InternalAffairs/UselessRestrictOnSend` and I'm not sure what's the best way to handle that?

<img width="688" alt="image" src="https://github.com/user-attachments/assets/c1b32224-968e-4882-8894-fd89f5007482">

Reply:
There are only two other mixins that use `RESTRICT_ON_SEND`, and both of them actually define `on_send` as well. I think there's a few options here:

1. Inline disable `InternalAffairs/UselessRestrictOnSend` here.
2. Update this class to implement `on_send` as well, and further refactor out `HashSlice` and `HashExcept`.  This should be plausible because other than `return unless semantically_slice_method?(node)` or `return unless semantically_except_method?(node)`, the `on_send` method in both cops are essentially the same.